include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -2731,6 +2731,22 @@ class FEFaceEvaluation : public FEEvaluationAccess<dim,
   std_cxx20::ranges::iota_view<unsigned int, unsigned int>
   dof_indices() const;
 
+  /**
+   * Return whether the face is at the boundary.
```

Comment:
I am not entirely convinced yet of the proposed function at this place, which is exemplified by how you write the comment. In reality, the present object does not represent a face, but it is an evaluation object that is *initialized to operate on a face*. Hence, I feel that the name `at_boundary()` or `boundary_id()` use elsewhere in the library does not really tell that this property depends on the state of the object. What should happen if the evaluation object has not yet been initialized?

What is the intended use case? I guess the use case is that you hand in the `FEFaceEvaluation` object to another function, which would then make it cumbersome to figure out where the face id (face batch id) came from?

As a final remark, I would like to express concerns that the function `boundary_id` duplicates information about `MatrixFree::get_boundary_id()`, so one would need to ask if there are additional functions for which we could provide a similar intuitive interface or if we can already foresee that the old interface would be deprecated one day. I am reluctant to make these kind of changes unless we have a clear plan towards something that is consistently better. This does not mean that I would be opposed to adding this interface, but I think this needs to be well thought-out and it needs to be documented in an intuitive way.

Reply:
> What is the intended use case? I guess the use case is that you hand in the FEFaceEvaluation object to another function, which would then make it cumbersome to figure out where the face id (face batch id) came from?

Yes. Once `FEFaceEvaluation::reinit()` has been called, I would like that user do not have to use `MatrixFree` and `face_index` directly anymore to reduce potential reason for error. In the worst case, one has to write `phi.get_matrix_free().get_boundary_id(phi.get_matrix_free.get_cell_or_face_batch_id());`.

> What should happen if the evaluation object has not yet been initialized?

I'll add an assert using `is_reinitialized`.

> As a final remark, I would like to express concerns that the function boundary_id duplicates information about MatrixFree::get_boundary_id(), so one would need to ask if there are additional functions for which we could provide a similar intuitive interface or if we can already foresee that the old interface would be deprecated one day. I am reluctant to make these kind of changes unless we have a clear plan towards something that is consistently better. This does not mean that I would be opposed to adding this interface, but I think this needs to be well thought-out and it needs to be documented in an intuitive way.

I know that this duplicates the ways to access the same information and I wouldn't suggest to deprecate the other one. I see both legitimate and the usage depending on the context. In the case when someone is in a matrix-free loop and operates on faces, one uses the new one. In the case when someone is setting up some data structures that are later on accessed during a matrix-free loop, the old one is used. In this context one could create a dummy `FEFaceEvaluation` object and call the new function but that seems a bit inconvenient. 

Similarly, we also have the `read/set_cell_data()` functions, which only contain bound checks and hide the usage of `cell_index`/`face_index`. In the case of the face versions, the situation is a bit more complicated.

I have chosen the names to match to what is in `TriaAccessor` so that one does not have get used to new names. But, if we decide to go forward, we can also give other names.

Reply:
@kronbichler Does my reasoning make sense to you? If there are still doubts, I will remove the usage in https://github.com/dealii/dealii/pull/14121 so that that PR can progress.

Reply:
I see. I think we can proceed, and I do not have better suggestions for names. Anyway, we need to spend more time on writing better comments that explain this thoroughly and make sure the user's intuition is met.

Reply:
@kronbichler I have updated the documentation.